### PR TITLE
Allow creation of API keys for App Insights

### PR DESCRIPTION
**Pull request by frodopwns**
**What this PR does / why we need it**:
Allows for the creation of App Insights API keys. These keys allow applications to read telemetry data or write annotations to metrics charts.

**Special notes for your reviewer**:
Create an App Insights, then create an App Insights API Key.
Go to the App insights in the portal and look to the left column for API Access. Should see the key there.

Also a secret should be created with the api key in it.

### CODE REVIEW

**Comment by frodopwns** [comment]
thanks @jananivMS 

Addressed the stack trace for 404s from creating the key. Also repackaged the helm chart.

File: pkg/resourcemanager/appinsights/api_keys_reconcile.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+		opt(options)
+	}
+
+	if options.SecretClient != nil {
+		c.SecretClient = options.SecretClient
+	}
+
+	instance.Status.Provisioning = true
+
+	// we have reconciled this previously, check if it already exists
```
Comment (theunrepentantgeek) [inline pkg/resourcemanager/appinsights/api_keys_reconcile.go:40]:
```suggestion
	// we may have reconciled this previously, check if it already exists
```

File: pkg/resourcemanager/appinsights/api_keys_reconcile.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+			instance.Spec.AppInsights,
+			idParts[len(idParts)-1],
+		)
+		if err == nil {
+			return true, nil
+		}
+
+		return false, nil
+
+	}
```
Comment (theunrepentantgeek) [inline pkg/resourcemanager/appinsights/api_keys_reconcile.go:56]: Remove extra blank lines #trivial-nit

File: pkg/resourcemanager/appinsights/api_keys_reconcile.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+			instance.Spec.ResourceGroup,
+			instance.Spec.AppInsights,
+			idParts[len(idParts)-1],
+		)
+		if err == nil {
+			return true, nil
+		}
+
+		return false, nil
+
```
Comment (frodopwns) [inline pkg/resourcemanager/appinsights/api_keys_reconcile.go:55]:
```suggestion
```

File: pkg/resourcemanager/appinsights/api_keys_reconcile.go (hunk 2 above)
Comment (frodopwns) [reply pkg/resourcemanager/appinsights/api_keys_reconcile.go:55]: done...I think there may be a better go fmt lib that can automate this...

**Pull request merged**